crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -293,6 +314,68 @@ impl Timeline {
         }
     }
 
+    /// Send a reply to the given event.
+    ///
+    /// Currently it only supports events with an event ID and JSON being
+    /// available (which can be removed by local redactions). This is subject to
+    /// change. Please check [`EventTimelineItem::can_be_replied_to`] to decide
+    /// whether to render a reply button.
+    ///
+    /// The sender of `reply_item` will be added to the mentions of the reply if
+    /// and only if `reply_item` has not been written by the sender.
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content of the reply
+    ///
+    /// * `event_id` - The event id of the item you want to reply to
+    ///
+    /// * `forward_thread` - Usually `Yes`, unless you explicitly want to the
+    ///   reply to show up in the main timeline even though the `reply_item` is
+    ///   part of a thread
+    pub async fn send_reply_with_event(
```

Comment:
Same high-level comment for this function and for the edit part.

Instead of having two methods:

- one to send using an event id
- one to send using an event timeline item

Rather, I was thinking about the following:

- one method to retrieve the `ReplyInfo` from an event id
- one method to retrieve the `ReplyInfo` from an event timeline item (would return an Option, if it's been found)
- and the `send_reply` method you already have

This would make the API leaner, and then only the FFI layer could use the two first methods in combination to get a `ReplyInfo`, and then use it in `send_reply`.

And same for the edit.

Reply:
Okay I see this means I need to expose `ReplyInfo` and `EditInfo` also through FFI right?

Reply:
`ReplyInfo` and `EditInfo` would need to be exposed outside of the SDK UI crate, but then the FFI could make use of those internally to provide the final functionality the apps require.